django_project/api/api_views/v2/authentication.py

```diff
@@ -12,10 +12,31 @@ def authenticate(self, request):
         if not api_key:
             raise exceptions.AuthenticationFailed('api-key is needed')
         key = UserApiKey.get_key_from_api_key(api_key)
+        if not key:
+            raise exceptions.AuthenticationFailed(
+                'api-key is invalid, '
+                'please recreate using enrollment form on profile page.'
+            )
+        if not key.is_active:
+            raise exceptions.AuthenticationFailed(
+                'api-key is not active, please wait for admin approval.'
+            )
         user = key.user
         if not user:
             raise exceptions.AuthenticationFailed('api-key is invalid')
         if request.method != 'GET' and not key.allow_write:
-            raise exceptions.AuthenticationFailed('this api-key is not allowed to post data')
+            raise exceptions.AuthenticationFailed(
+                'This api-key is not allowed to post data. '
```

Comment:
This API key does not include access permissions to post data.                                                             Please ask the admins at https://healthsites.io/ if you need POST permissions.